asf_search/ASFSession.py

```diff
@@ -72,11 +73,22 @@ def auth_with_creds(self, username: str, password: str):
         login_url = f'https://{self.edl_host}/oauth/authorize?client_id={self.edl_client_id}&response_type=code&redirect_uri=https://{self.asf_auth_host}/login'
 
         self.auth = (username, password)
+
+        ASF_LOGGER.info(f'Attempting to login via "{login_url}"')
         self.get(login_url)
 
         if not self._check_auth_cookies(self.cookies.get_dict()):
             raise ASFAuthenticationError("Username or password is incorrect")
 
+        ASF_LOGGER.info(f'Login successful')
+        token = self.cookies.get_dict().get('urs-access-token')
+        if token is not None:
+            ASF_LOGGER.info(f'Acquired valid application bearer token, ASFSession will use this to validate future queries (required for accessing authorized hidden/restricted datasets)')
+            self.auth = None
+            self._update_edl_token(token=token)
+        else:
+            ASF_LOGGER.error(f"Configured ASFSession Auth Host {self.asf_auth_host} failed to return a user token. A valid token is required for accessing hidden/restricted datasets")
+        
```

Comment:
Should this error out too after logging error?

Reply:
Went ahead and just made this raise an error!